pallets/automation-time/src/fees.rs

```diff
@@ -71,59 +78,110 @@ where
+				// we need to add the fees to check whether they are sufficient,
+				// otherwise check them separately.
+				let exec_fee_location = exec_fee
+					.asset_location
+					.reanchored(&T::SelfLocation::get(), T::UniversalLocation::get())
+					.map_err(|_| Error::<T>::CannotReanchor)?;
+
+				let schedule_fee_location = self
+					.schedule_fee
+					.asset_location
+					.reanchored(&T::SelfLocation::get(), T::UniversalLocation::get())
+					.map_err(|_| Error::<T>::CannotReanchor)?;
+
+				if exec_fee_location == schedule_fee_location {
+					let fee = self.schedule_fee.amount.saturating_add(exec_fee.amount);
+					Self::ensure_can_withdraw(self, exec_fee.asset_location, fee)?;
+				} else {
+					Self::ensure_can_withdraw(
+						self,
+						self.schedule_fee.asset_location,
+						self.schedule_fee.amount,
+					)?;
+				}
+			},
+			_ => {
+				Self::ensure_can_withdraw(
+					self,
+					self.schedule_fee.asset_location,
+					self.schedule_fee.amount,
+				)?;
+			},
+		}
+
 		Ok(())
 	}
 
 	/// Withdraw the fee.
 	fn withdraw_fee(&self) -> Result<(), DispatchError> {
-		let fee = self.schedule_fee_amount.saturating_add(self.execution_fee_amount);
+		log::debug!(target: "FeeHandler", "FeeHandler::withdraw_fee, self.schedule_fee.asset_location: {:?}, self.schedule_fee.amount: {:?}",
+			self.schedule_fee.asset_location, self.schedule_fee.amount);
+		// Withdraw schedule fee
+		if !self.schedule_fee.amount.is_zero() {
+			let currency_id = T::CurrencyIdConvert::convert(self.schedule_fee.asset_location)
+				.ok_or("InconvertibleMultilocation")?;
 
-		if fee.is_zero() {
-			return Ok(())
-		}
+			T::MultiCurrency::withdraw(currency_id.into(), &self.owner, self.schedule_fee.amount)
+				.map_err(|_| DispatchError::Token(BelowMinimum))?;
 
-		let currency_id = T::CurrencyIdConvert::convert(self.schedule_fee_location)
-			.ok_or("IncoveribleMultilocation")?;
-		let currency_id = currency_id.into();
+			TR::take_revenue(MultiAsset {
+				id: AssetId::Concrete(self.schedule_fee.asset_location),
+				fun: Fungibility::Fungible(self.schedule_fee.amount.saturated_into()),
+			});
+		}
 
-		match T::MultiCurrency::withdraw(currency_id, &self.owner, fee) {
-			Ok(_) => {
-				TR::take_revenue(MultiAsset {
-					id: AssetId::Concrete(self.schedule_fee_location),
-					fun: Fungibility::Fungible(self.schedule_fee_amount.saturated_into()),
-				});
+		// Withdraw execution fee
+		if let Some(execution_fee) = &self.execution_fee {
+			if execution_fee.is_local {
+				log::debug!(target: "FeeHandler", "FeeHandler::withdraw_fee, self.execution_fee.asset_location: {:?}, self.execution_fee.amount: {:?}",
+					execution_fee.asset_location, execution_fee.amount);
+				let currency_id = T::CurrencyIdConvert::convert(execution_fee.asset_location)
+					.ok_or("InconvertibleMultilocation")?;
 
-				if self.execution_fee_amount > MultiBalanceOf::<T>::zero() {
+				let execution_fee_amount = execution_fee.amount;
+				if !execution_fee_amount.is_zero() {
```

Comment:
The same question as above.

Reply:
When the execution_fee_amount is not equal to zero, execute the withdrawal process; otherwise, skip it.

Reply:
Please add your explanation to this section of code as a comment.

```
When the expected deduction amount, execution_fee_amount, is not equal to zero, execute the withdrawal process; otherwise, there’s no need to deduct.
```